### PR TITLE
Fixes #30406: default_capsule method exists on SmartProxy object

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: app/services/default_proxy_proxy_selector.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ def initialize
   def available_proxies(host, provider)
     # TODO: Once we have a internal proxy marker/feature on the proxy, we can
     # swap the implementation
-    internal_proxy = ::Katello.default_capsule
+    internal_proxy = ::SmartProxy.default_capsule
```
Comment (ezr-ondrej) [inline app/services/default_proxy_proxy_selector.rb:13]: It's still comming from a Katello, so this call still implies dependency on `Katello` just less explicitly.
Should we instead implement notion of `internal_proxy` in the Foreman itself?

**Comment by ehelms** [comment]
The method is defined on the smart proxy just within Katellos code. The
dependency is still there on Katello.

**Comment by ezr-ondrej** [comment]
> The method is defined on the smart proxy just within Katellos code. The dependency is still there on Katello.

Yeah, that's what I'm saying :thinking: but with calling it on `Katello` it's obvious it's comming from katello, but like this I'd not be sure. So it doesn't improve the situation much IMHO.

**Comment by ehelms** [comment]
I can add code that will do `fail("default_capsule must be defined") unless  ::SmartProxy.defined?('default_capsule')`

**Comment by adamruzicka** [comment]
It would be better than having it fail with undefined method error in case katello is not available

**Comment by ehelms** [comment]
Note https://github.com/theforeman/foreman_remote_execution/pull/519/files#diff-84297365d8d19c4d3bfa8769908bbb85R5 protects against missing Katello. But I will add another layer.

**Comment by ezr-ondrej** [comment]
Rubocop doesn't like `fail` I'd agree `raise` is what we want.

**Comment by adamruzicka** [comment]
Thank you @ehelms & @ezr-ondrej !

**Pull request merged**